profiler/profile.go

```diff
@@ -319,6 +341,157 @@ func (d *deltaProfiler) Delta(curData []byte) ([]byte, error) {
 	return deltaData, nil
 }
 
+type fastDeltaProfiler struct {
+	dc  *fastdelta.DeltaComputer
+	buf bytes.Buffer
+	gzr gzip.Reader
+	gzw *gzip.Writer
+}
+
+func newFastDeltaProfiler(v ...pprofutils.ValueType) deltaProfiler {
+	fd := &fastDeltaProfiler{
+		dc: fastdelta.NewDeltaComputer(v...),
+	}
+	fd.gzw = gzip.NewWriter(&fd.buf)
+	return fd
+}
+
+func isGzipData(data []byte) bool {
+	// the first two bytes of gzip data are 0x1f8b
+	return len(data) >= 2 && (data[0] == 0x1f && data[1] == 0x8b)
+}
+
+func (fdp *fastDeltaProfiler) Delta(data []byte) (b []byte, err error) {
+	if isGzipData(data) {
+		if err := fdp.gzr.Reset(bytes.NewReader(data)); err != nil {
+			return nil, err
+		}
+		data, err = io.ReadAll(&fdp.gzr)
+		if err != nil {
+			return nil, fmt.Errorf("decompressing profile: %v", err)
+		}
+	}
+
+	fdp.buf.Reset()
+	fdp.gzw.Reset(&fdp.buf)
+
+	if err = fdp.dc.Delta(data, fdp.gzw); err != nil {
+		return nil, fmt.Errorf("error computing delta: %v", err)
+	}
+	if err = fdp.gzw.Close(); err != nil {
+		return nil, fmt.Errorf("error flushing gzip writer: %v", err)
+	}
+	return fdp.buf.Bytes(), nil
+}
```

Comment:
This was something I asked for when fastdelta was its own library.
Compression is an orthogonal concern to the process of computing delta's.
You are right `pprofile.ParseData()` dealt with it.  Keeping that out of DeltaComputer is exactly because it is not at the same level of abstraction as `pprofile.ParseData()` and `deltaProf.Write()`.

Reply:
Ok, I don't feel strongly about it.